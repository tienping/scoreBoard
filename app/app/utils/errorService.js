// this is service where all the error catching will be store
//
// example: if got try and catch can have error log message and object that return in current process.
//
// the parameter are error text (needed) and object (optional)

exports.errorLog = function (text, obj = null) {
    console.log(text, obj);
};
