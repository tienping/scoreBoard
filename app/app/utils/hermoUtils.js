/**
 * App Utilities
 */
import _ from 'lodash';
import { create } from 'apisauce';
import { NativeModules, Platform, Dimensions, PixelRatio, ToastAndroid } from 'react-native';
import { globalScope } from 'hermo/globalScope';

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

const addHeaderToAPI = (apiString, headerParams) => {
    const api = create({
        baseURL: apiString,
        headers: {
            'Content-Type': 'application/json',
            'Accept-Language': 'en',
            'api-version': '1.0.0',
            // 'app-version': DeviceInfo.getVersion(),
            'app-os-name': Platform.OS,
            'hertoken': globalScope.token,
            // 'app-os-version': DeviceInfo.getSystemVersion(),
        },
        timeout: 30000,
    });

    if (headerParams) {
        api.addAsyncRequestTransform((request) => async () => {
            NativeModules.RNAttributesService.getApiHeaderAttributes((error, headers) => {
                request.headers = { ...request.headers, ...headerParams };
            });
        });
    }

    return api;
};

export function mockupHeader() {
    const headerParams = {
        'x-hermo-one-signal-id': 'value',
        'x-hermo-one-device-token': 'value',
        'x-hermo-one-ads-id': 'value',
        'x-hermo-one-session-id': 'value',
        'x-hermo-one-app-id': 'value',
    };
    const apiObject = addHeaderToAPI(globalScope.api, headerParams);
    return apiObject.post('/');
}

export function apiRequest(path, type, body, url, headerParams) {
    if (!globalScope.isConnected) {
        herlert('Please check your connection.');
    }
    const apiObject = addHeaderToAPI(url || globalScope.api);
    return apiObject[type](path, body, headerParams);
}

export function staticErrorResponse(message) {
    return {
        success: false,
        messages: [message],
    };
}

export function dataChecking(object) {
    const args = Array.prototype.slice.call(arguments, 1);
    let obj = object;

    for (let i = 0; i < args.length; i += 1) {
        if (!obj || !Object.prototype.hasOwnProperty.call(obj, args[i])) {
            return null;
        }
        obj = obj[args[i]];
    }
    return obj;
}

export function getBannerHeight(imageProperty, width) {
    const imageHeight = imageProperty && imageProperty.height ? imageProperty.height : 280;// dataChecking(imageProperty, height)
    const imageWidth = imageProperty && imageProperty.width ? imageProperty.width : 560;

    return ((imageHeight / imageWidth) * width);
}

export function shuffleData(data) {
    return _.shuffle(data);
}

// to convert the  percentage of width to the dp;
// eg: getXdp(50)
export const getXdp = (percent) => PixelRatio.roundToNearestPixel((screenWidth * parseFloat(percent)) / 100);
// to convert the  percentage of height to the dp;
// eg: getYdp(50)
export const getYdp = (percent) => PixelRatio.roundToNearestPixel((screenHeight * parseFloat(percent)) / 100);

export function transfromViewWidthToPercentage(percentage) {
    return screenWidth * percentage;
}

export function transfromViewHeightToPercentage(percentage) {
    return screenHeight * percentage;
}

export function transfromViewHeightBasedOnColumn(sourceHeight, sourceWidth, column) {
    const viewWidth = screenWidth / column;
    const ratio = sourceHeight / sourceWidth;
    return viewWidth * ratio;
}

export function transfromViewHeightBasedOnColumnNScreenWidth(sWidth, sourceHeight, sourceWidth, column) {
    const viewWidth = sWidth / column;
    const ratio = sourceHeight / sourceWidth;
    return viewWidth * ratio;
}

export function transfromViewHeightBasedOnPercentage(sourceHeight, sourceWidth, percentage) {
    const viewWidth = screenWidth * percentage;
    const ratio = sourceHeight / sourceWidth;
    return viewWidth * ratio;
}

export function getFullScreenWidth() {
    return screenWidth;
}

export function getFullScreenHeight() {
    return screenHeight;
}

export function appLinkGenerator(target, page, href, id) {
    const applink = {
        target,
        type: '',
        id,
        page: {
            id: page,
            params: [
                {
                    key: 'url',
                    value: href,
                },
            ],
        },
    };

    return applink;
}

export const HerToast = {
    show: ({ text, type, duration, navigator }) => {
        if (Platform.OS === 'ios') {
            // type: info / warning / success / error
            const toastType = type === 'danger' ? 'error' : type;
            NativeModules.ToastHandler.alert(null, text, toastType);
        } else {
            navigator.showSnackbar({
                text,
                // actionText: 'done', // optional
                // actionId: 'fabClicked', // Mandatory if you've set actionText
                // actionColor: 'green', // optional
                textColor: 'white', // optional
                backgroundColor: (type === 'danger') ? 'red' : 'green', // optional
                duration: duration || 'short', // default is `short`. Available options: short, long, indefinite
            });
        }
    },
};

export function herlert(string) {
    if (Platform.OS === 'ios') {
        NativeModules.ToastHandler.alert(null, string, 'info');
    } else {
        ToastAndroid.show(string, ToastAndroid.LONG);
    }

    return true;
}
