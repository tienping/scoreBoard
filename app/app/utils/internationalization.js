import I18n, { getLanguages } from 'react-native-i18n';

I18n.fallbacks = true;
I18n.locale = 'en'; // TODO: on language input setting, set into realm and get value on init

I18n.translations = {
    en: {
        locale: `Default Locale: ${I18n.currentLocale()}`,
    },
    zn: {
        locale: `初始语言: ${I18n.currentLocale()}`,
    },
};

// get user preferred locales
// getLanguages().then((languages) => {
//     console.log(languages); // ['en-US', 'en']
// });

export default I18n;
