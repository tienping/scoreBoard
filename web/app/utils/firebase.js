import * as firebase from 'firebase';
import { FirebaseConfig } from './firebaseConfig';

firebase.initializeApp(FirebaseConfig);

export const databaseRef = firebase.database().ref();
export const dataRef = databaseRef.child("data");

export function getFirebaseSnapshot(ref) {
    return new Promise((resolve) => {
        ref.on('value', function(snapshot) {
            if (snapshot.exists()) {
                resolve(snapshot.val());
            }
            resolve(null);
        });
    });
}