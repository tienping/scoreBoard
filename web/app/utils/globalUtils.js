export function getURLParams(urlparams) {
    const result = {};
    let param = urlparams;
    if (typeof param === 'string') {
        param = param.substr(1).split('&').forEach((str) => {
            const d = str.split('=');
            result[d[0]] = decodeURIComponent(d[1]);
        });
    }

    return result;
}

export function staticErrorResponse(message) {
    return {
        success: false,
        messages: [message],
    };
}
