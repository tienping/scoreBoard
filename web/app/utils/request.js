import 'whatwg-fetch';
import { sessionService } from 'redux-react-session';

/**
 * Parses the JSON returned by a network request
 *
 * @param  {object} response A response from a network request
 *
 * @return {object}          The parsed JSON from the request
 */
function parseJSON(response) {
    if (response.status === 204 || response.status === 205) {
        return null;
    }

    try {
        return response.json();
    } catch (e) {
        return {
            success: false,
            messages: [{
                text: 'Please check your internet connection',
            }],
        };
    }
}

/**
 * Requests a URL, returning a promise
 *
 * @param  {string} url       The URL we want to request
 * @param  {object} [options] The options we want to pass to "fetch"
 *
 * @return {object}           The response data
 */
export default function request(url, options) {
    return fetch(url, options)
        .then(parseJSON)
        .catch(parseJSON);
}

export function loadSession() {
    return sessionService.loadSession()
        .then((data) => (data));
        // Catch was removed here
        // it should be
        // .catch((err) => {
        //     throw err;
        // });
}
