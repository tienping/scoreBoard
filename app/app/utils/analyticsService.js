import firebase from 'react-native-firebase';
const Fabric = require('react-native-fabric');
const ReactMoE = require('react-native-moengage');
const { Answers } = Fabric;

firebase.analytics().setAnalyticsCollectionEnabled(true);

export const FirebaseAnalytics = {
    logAddToWishList: ({ item_id, item_name, item_category, item_location_id, value, price, currency }) => {
        firebase.analytics().logEvent('add_to_wishlist', { item_id, item_category, item_name, item_location_id, value, price, currency });
    },
    logAddToCart: ({ item_id, item_name, item_category, item_location_id, value, price, currency, quantity }) => {
        firebase.analytics().logEvent('add_to_cart', { item_id, item_category, item_name, item_location_id, value, price, currency, quantity });
    },
    logSearch: ({ search_text }) => {
        firebase.analytics.logEvent('search', { search_text });
    },
    logPageView: ({ id, name, sale_price, page_type }) => {
        firebase.analytics().logEvent('page_view', { id, name, sale_price, page_type });
    },
    logCheckoutSuccess: ({ order_id, order_amount, estimated_delivery_date, delivery_medium }) => {
        firebase.analytics().logEvent('checkout_success', { order_id, order_amount, estimated_delivery_date, delivery_medium });
    },
    logCheckoutError: ({ order_id, order_amount, error_code, error_description, retry_number }) => {
        firebase.analytics().logEvent('checkout_success', { order_id, order_amount, error_code, error_description, retry_number });
    },
    logInstall: ({ source_of_install }) => {
        firebase.analytics().logEvent('install', { source_of_install });
    },
    logStartedSignUp: ({ source_of_install }) => {
        firebase.analytics().logEvent('started_signup', { source_of_install });
    },
    logSignUpSubmit: ({ email, phone, method }) => {
        firebase.analytics().logEvent('signup_submit', { email, phone, method });
    },
    logSignUpSuccess: ({ email, phone, method, FbID }) => {
        firebase.analytics().logEvent('signup_success', { email, phone, method, FbID });
    },
    logSignUpFailed: ({ email, phone, errCode, errorDescription }) => {
        firebase.analytics().logEvent('signup_success', { email, phone, errCode, errorDescription });
    },
};

export const FabricAnalytics = {
    logAddToWishList: ({ id, brand, name, price, status, pagePosition }) => {
        Answers.logCustom('Wish List', {
            'Product Id': id,
            'Product Name': name,
            'Product Brand': brand,
            'Product Price': price,
            'Event Success': status,
            'Event Triggered From': pagePosition,
        });
    },
    logAddToCart: ({ id, type, brand, name, price, qty, status, pagePosition }) => {
        Answers.logCustom('Add To Cart', {
            'Product Id': id,
            'Product Type': type,
            'Product Name': name,
            'Product Brand': brand,
            'Product Price': price,
            'Product Qty': qty,
            'Event Success': status,
            'Event Triggered From': pagePosition,
        });
    },
    logDynamicView: ({ id, url, pagePosition }) => {
        Answers.logCustom('Scroll', {
            'Page Id': id,
            'Page Url': url,
            'Event Triggered From': pagePosition,
        });
    },
    logProductView: ({ id, brand, name, price, pagePosition }) => {
        Answers.logCustom('Product View', {
            'Product Id': id,
            'Product Name': name,
            'Product Brand': brand,
            'Product Price': price,
            'Event Triggered From': pagePosition,
        });
    },
    logActivityView: ({ page_type }) => {
        Answers.logCustom('Activity View', { 'Page Type': page_type });
    },
    logTimetableView: ({ type, url, pagePosition }) => {
        Answers.logCustom('Timetable View', {
            'Timetable Type': type,
            'Timetable Links': url,
            'Event Triggered From': pagePosition,
        });
    },
    logMakeOrder: ({ time, courier, gateway, total, shipping, status, pagePosition }) => {
        Answers.logCustom('Make Order', {
            'Order Created Time': time,
            'Order Courier': courier,
            'Order Payment Gateway': gateway,
            'Order Subtotal': total,
            'Order Shipping Value': shipping,
            'Event Success': status,
            'Event Triggered From': pagePosition,
        });
    },
    logMakePayment: ({ gateway, paymentChannel, total, time, status, pagePosition }) => {
        Answers.logCustom('Make Payment', {
            'Payment Gateway': gateway,
            'Payment Channel': paymentChannel,
            'Payment Amount': total,
            'Payment Created At': time,
            'Payment Status': status,
            'Event Triggered From': pagePosition,
        });
    },
    logScroll: ({ pagination, pagePosition }) => {
        Answers.logCustom('Scroll', {
            'Pagination': pagination,
            'Event Triggered From': pagePosition,
        });
    },
    logMainTabbed: ({ tab, pagePosition }) => {
        Answers.logCustom('Main Tabbed', {
            'Page Type': tab,
            'Event Triggered From': pagePosition,
        });
    },
    logSearch: ({ term, result }) => {
        Answers.logCustom('Search', {
            'Search String': term,
            'Search Result Found': result,
        });
    },
    logShare: ({ shareType, shareValue, pagePosition }) => {
        Answers.logCustom('Share', {
            'Share Type': shareType,
            'Share Value': shareValue,
            'Event Triggered From': pagePosition,
        });
    },
    logButtonClick: ({ buttonName, pagePosition }) => {
        Answers.logCustom('Button Click', {
            'Button Name': buttonName,
            'Event Triggered From': pagePosition,
        });
    },
    logSignUp: ({ type }) => {
        Answers.logCustom('Sign up', {
            'SignUp Type': type,
        });
    },
    logLogin: ({ type }) => {
        Answers.logCustom('Login', {
            'Login Type': type,
        });
    },
    logBeautyWallShare: ({ value }) => {
        Answers.logCustom('Share Beauty Review', {
            'Share Value': value,
        });
    },
    logBeautyReviewOpen: ({ value }) => {
        Answers.logCustom('Click Beauty Review Link', {
            'Share Value': value,
        });
    },
    logBeautyReviewConverted: ({ type }) => {
        Answers.logCustom('Beauty Review Converted', {
            'Converted Type': type,
        });
    },
};

export const MoEngageAnalytics = {
    // ReactMoE.trackEvent("Purchase", {"quantity":1, "product":"iPhone", "currency":"dollar", "price":699, "new_item" : true});
    setUniqueID: ({ id }) => {
        ReactMoE.setUserUniqueID(id);
    },
    setUserAttributes: ({ email, gender, ethnicity, name, membership }) => {
        ReactMoE.setUserAttribute(email);
        ReactMoE.setUserAttribute(gender);
        ReactMoE.setUserAttribute(ethnicity);
        ReactMoE.setUserAttribute(name);
        ReactMoE.setUserAttribute(membership);
    },
    logSearch: ({ search_text }) => {
        ReactMoE.trackEvent('analytic_search', { 'search_text': search_text });
    },
    logPageView: ({ id, name, type }) => {
        ReactMoE.trackEvent('analytic_pageView', {
            'id': id,
            'name': name,
            'page_type': type,
        });
    },
    logProductView: ({ id, name, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price }) => {
        ReactMoE.trackEvent('analytic_productView', {
            'id': id,
            'name': name,
            'price': price,
            'group_id': groupId,
            'group_name': groupName,
            'category_id': categoryId,
            'category_name': categoryName,
            'sub_category_id': subCategoryId,
            'sub_category_name': subCategoryName,
        });
    },
    logAddtoCart: ({ id, name, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price, quantity }) => {
        ReactMoE.trackEvent('analytic_addToCart', {
            'id': id,
            'name': name,
            'price': price,
            'group_id': groupId,
            'quantity': quantity,
            'group_name': groupName,
            'category_id': categoryId,
            'category_name': categoryName,
            'sub_category_id': subCategoryId,
            'sub_category_name': subCategoryName,
        });
    },
    logRemoveFromCart: ({ id, name, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price, quantity }) => {
        ReactMoE.trackEvent('analytic_removeFromCart', {
            'id': id,
            'name': name,
            'price': price,
            'group_id': groupId,
            'quantity': quantity,
            'group_name': groupName,
            'category_id': categoryId,
            'category_name': categoryName,
            'sub_category_id': subCategoryId,
            'sub_category_name': subCategoryName,
        });
    },
    logAddToWishList: ({ id, name, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price }) => {
        ReactMoE.trackEvent('analytic_addToWishlist', {
            'id': id,
            'name': name,
            'price': price,
            'group_id': groupId,
            'group_name': groupName,
            'category_id': categoryId,
            'category_name': categoryName,
            'sub_category_id': subCategoryId,
            'sub_category_name': subCategoryName,
        });
    },
    logRemoveFromWishList: ({ id, name, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price }) => {
        ReactMoE.trackEvent('analytic_removeFromWishlist', {
            'id': id,
            'name': name,
            'price': price,
            'group_id': groupId,
            'group_name': groupName,
            'category_id': categoryId,
            'category_name': categoryName,
            'sub_category_id': subCategoryId,
            'sub_category_name': subCategoryName,
        });
    },
    logOrderCreated: ({ orderId, cartSize, cartAmount, purchaseAmount, productId, productName, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price, quantity, city, state, postcode, pcodeApplied, pcodeName, pcodeAmount, payment, courier, estimateDelivery }) => {
        ReactMoE.trackEvent('analytic_orderCreated', {
            'price': price,
            'payment': payment,
            'courier': courier,
            'order_id': orderId,
            'group_id': groupId,
            'quantity': quantity,
            'address_city': city,
            'cart_size': cartSize,
            'address_state': state,
            'product_id': productId,
            'pcode_name': pcodeName,
            'group_name': groupName,
            'category_id': categoryId,
            'cart_amount': cartAmount,
            'product_name': productName,
            'pcode_amount': pcodeAmount,
            'address_postcode': postcode,
            'category_name': categoryName,
            'pcode_applied': pcodeApplied,
            'sub_ategory_id': subCategoryId,
            'purchase_amount': purchaseAmount,
            'sub_category_name': subCategoryName,
            'estimate_delivery': estimateDelivery,
        });
    },
    logPaymentSuccess: ({ orderId, cartSize, cartAmount, purchaseAmount, productId, productName, groupId, groupName, categoryId, categoryName, subCategoryId, subCategoryName, price, quantity, city, state, postcode, pcodeApplied, pcodeName, pcodeAmount, payment, courier, estimateDelivery }) => {
        ReactMoE.trackEvent('analytic_paymentSuccess', {
            'price': price,
            'payment': payment,
            'courier': courier,
            'order_id': orderId,
            'group_id': groupId,
            'quantity': quantity,
            'address_city': city,
            'cart_size': cartSize,
            'address_state': state,
            'product_id': productId,
            'pcode_name': pcodeName,
            'group_name': groupName,
            'category_id': categoryId,
            'cart_amount': cartAmount,
            'product_name': productName,
            'pcode_amount': pcodeAmount,
            'address_postcode': postcode,
            'category_name': categoryName,
            'pcode_applied': pcodeApplied,
            'sub_ategory_id': subCategoryId,
            'purchase_amount': purchaseAmount,
            'sub_category_name': subCategoryName,
            'estimate_delivery': estimateDelivery,
        });
    },
    logOut: () => {
        ReactMoE.logout();
    },
};
