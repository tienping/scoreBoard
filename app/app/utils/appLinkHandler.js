import { dataChecking, HerToast  } from 'utils/hermoUtils';
import branch from 'react-native-branch';
import { globalScope } from 'hermo/globalScope';
import { Platform, AlertIOS, ToastAndroid } from 'react-native';
const Clipboard = require('react-native-clipboard');

import { store } from '../../App';

const TARGET_IN_PAGE = 'in-page';
const TARGET_NEW_PAGE = 'new-page';
const TYPE_MALL = 'mall';
const TYPE_MASK = 'mask';
const TYPE_BRAND = 'brand';
const TYPE_GROUP = 'group';
const TYPE_CATEGORY = 'category';
const TYPE_SUBCATEGORY = 'subcategory';
const TYPE_PROMOTION = 'promotion';
const TYPE_REVIEW_MALL = 'review_mall';
const TYPE_EVENT_NOTIFY = 'event_notify';
const TYPE_SPECIAL_PAGE = 'special_page';

// new page_id
const PAGE_MALL_LIST = 'mall-list';
const PAGE_MALL_VIEW = 'mall-view';
const PAGE_MASK_VIEW = 'mask-view';
const PAGE_BUNDLE_VIEW = 'bundle-view';
const PAGE_MASK_CART = 'mask-cart';
const PAGE_HOTDEAL = 'hotdeal-list';
const PAGE_SEARCH_RESULT = 'search-result';
const PAGE_BEAUTY_GRAB = 'beauty-grab';
const PAGE_BEAUTY_WALL = 'beauty-wall';
const PAGE_EVENT_LIST = 'event-list';
const REVIEW_MALL = 'review-mall';
const OPEN_WEB_URL = 'open-web-url';
const OPEN_APP_STORE = 'open-app-store';

// in-page id
const POPUP_IMAGE_DIALOG = 'image-dialog';
const HTML_STRING_DIALOG = 'html-string-dialog';
const HTML_URL_DIALOG = 'html-url-dialog';
const VIDEO_DIALOG = 'video-dialog';
export const HTML_API_DIALOG = 'html-api-dialog';
const COPY = 'copy-to-clipboard';
const QUICK_ADD_TO_CART = 'cart-mall';
const EVENT_REDEEM = 'event-redeem';
const MODAL = 'modal';
const SHARE = 'share';

export function handleAppLink(applink, navigator, pageTitle) {
    if (!applink) {
        return null;
    }
    if (isNewPage(applink.target) && navigator) {
        decideDestination(applink, navigator, pageTitle);
    } else {
        decideInPageAction(applink, navigator);
    }
    return null;
}

function isNewPage(target) {
    if (target === TARGET_NEW_PAGE) {
        return true;
    }
    return false;
}

function obsorbUrl(params) {
    for (let i = 0; i < params.length; i += 1) {
        if (params[i].key === 'url') {
            return params[i].value;
        }
    }
    return null;
}

function decideDestination(applink, navigator, pageTitle) {
    const pageId = applink.page.id;
    const { type, id, page } = applink;
    const url = obsorbUrl(applink.page.params);

    switch (pageId) {
        case PAGE_MALL_LIST:
            navigateToScreen({
                navigator,
                routeName: 'hermorn.screen.HerListing',
                params: { data: url, pageTitle, flatListId: `${type}-${id}` },
            });
            break;
        case PAGE_MALL_VIEW:
            if (id) {
                if (id === '4877') {
                    navigateToScreen({
                        navigator,
                        routeName: 'hermorn.screen.BundleBox',
                        params: { bundleID: id, pageTitle: 'Bundle Box', },
                    });
                } else {
                    navigateToScreen({
                        navigator,
                        routeName: 'hermorn.screen.ProductInfo',
                        params: { data: id, pageTitle },
                    });
                }
            }
            break;
        // case PAGE_MASK_VIEW:
        //     break;
        case PAGE_BUNDLE_VIEW:
            if (dataChecking(id)) {
                navigateToScreen({
                    navigator,
                    routeName: 'hermorn.screen.BundleBox',
                    params: { bundleID: id, pageTitle: 'Bundle Box' },
                });
            }
            break;
        // case PAGE_MASK_CART:
        //     break;
        case PAGE_HOTDEAL:
            // event view
            break;
        case PAGE_SEARCH_RESULT:
            break;
        case PAGE_BEAUTY_GRAB:
            break;
        case PAGE_BEAUTY_WALL:
            if (id) {
                navigator.push({
                    screen: 'hermorn.screen.Herfeed',
                    animated: true,
                    backButtonTitle: '',
                    title: 'BEAUTY WALL',
                    passProps: {
                        data: id,
                    },
                });
            }
            break;
        case PAGE_EVENT_LIST:
            if (id && url) {
                navigateToScreen({
                    navigator,
                    routeName: 'hermorn.screen.CampaignViewer',
                    params: { data: { id, url }, pageTitle },
                });
            } else if (id) {
                navigateToScreen({
                    navigator,
                    routeName: 'hermorn.screen.CampaignViewer',
                    params: { data: { id, url: null }, pageTitle },
                });
            }
            break;
        case OPEN_WEB_URL:
            break;
        case OPEN_APP_STORE:
            break;
        default:
            break;
    }
}

export function navigateToScreen({ navigator, routeName, params }) {
    navigator.push({
        title: params.pageTitle || '',
        backButtonTitle: '',
        screen: routeName,
        passProps: params,
    });


    switch (routeName) {
        case 'proceedToBrandList':
            // store.dispatch(NavigationActions.navigate({ routeName: 'Brand' }));
            break;
        default:
            return null;
    }

    return null;
}
export function navigateToLightBox({ navigator, routeName, params }) {
    navigator.showLightBox({
        screen: routeName,
        passProps: params,
    });

}

function decideInPageAction(applink, navigator) {
    const pageId = applink.page.id;
    const url = obsorbUrl(applink.page.params);
    // const { id } = applink;
    switch (pageId) {
        case POPUP_IMAGE_DIALOG:
            navigateToLightBox({
                navigator,
                routeName: 'hermorn.screen.PopupDialog',
                params: { item: url, isImage: true },
            });
            break;
        case HTML_STRING_DIALOG:
            navigateToLightBox({
                navigator,
                routeName: 'hermorn.screen.PopupDialog',
                params: { item: url, isHtml: true },
            });
            break;
        case VIDEO_DIALOG:
        case HTML_URL_DIALOG:
            navigateToLightBox({
                navigator,
                routeName: 'hermorn.screen.PopupDialog',
                params: { item: url, isUrl: true },
            });
            break;
        case HTML_API_DIALOG:
            navigateToLightBox({
                navigator,
                routeName: 'hermorn.screen.PopupDialog',
                params: { item: url, isHtml: true, fireAPI: true, type: HTML_API_DIALOG},
            });
            break;
        case COPY:
            if (dataChecking(applink, 'id')) {
                Clipboard.set(applink.id);
                HerToast.show({
                    text: `${applink.id} Copied.`,
                    type: 'success',
                    duration: 5000,
                });
            }
            break;
        case QUICK_ADD_TO_CART:
            break;
        case EVENT_REDEEM:
            AlertIOS.alert(url);
            break;
        case MODAL:
            // store.dispatch(hermodalOpen({ content: <Text>asdfasf</Text>, type: 'video' }));
            break;
        case SHARE:
            break;
        default:
            break;
    }
}

export async function proceedShareAction(type, item) {
    let url;
    let page;
    let title;
    let desc;
    let imageUrl;

    switch (type) {
        case 'product':
            url = `https://www.hermo.my${item.url}`;
            title = dataChecking(item, 'display_name') ? item.display_name : '';
            desc = dataChecking(item, 'brief') ? item.brief : '';
            imageUrl = dataChecking(item, 'image', 'large') ? item.image.large : '';
            page = PAGE_MALL_VIEW;
            break;
        case 'event':
            url = `https://www.hermo.my${item.url}`;
            title = dataChecking(item, 'title') ? item.title : '';
            desc = dataChecking(item, '_page', 'share', 'caption') ? item._page.share.caption : '';
            imageUrl = dataChecking(item, '_page', 'share', 'picture') ? item._page.share.picture : '';
            page = PAGE_EVENT_LIST;
            break;
        case 'wall':
            url = `https://www.hermo.my/wall/${item.id}`;
            title = dataChecking(item, 'username') ? item.username : '';
            desc = dataChecking(item, 'comment') ? item.comment : '';
            imageUrl = dataChecking(item, 'image', 'original') ? item.image.original : '';
            page = PAGE_BEAUTY_WALL;
            break;
        default:
            break;
    }
    const branchUniversalObject = await branch.createBranchUniversalObject(`${item.id}`, {
        canonicalUrl: url,
        locallyIndex: true,
        title,
        contentDescription: desc,
        contentImageUrl: imageUrl,
        contentIndexingMode: 'public',
        contentMetadata: {
            page,
            href: dataChecking(item, '_links', 'self', 'href') ? item._links.self.href : '',
            item_id: item.id,
        },
    });

    const shareOptions = { messageHeader: 'Look Inside', messageBody: 'Check this out!' };
    const linkProperties = { feature: 'share', channel: 'facebook' };
    const controlParams = {
        $fallback_url: url,
        $desktop_url: url,
        $ios_url: 'app.hermo.my',
        $android_url: 'app.hermo.my',
    };

    await branchUniversalObject.showShareSheet(shareOptions, linkProperties, controlParams);
}
