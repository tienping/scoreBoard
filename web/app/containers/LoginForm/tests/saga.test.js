/**
 * Login sagas
 */

/* eslint-disable redux-saga/yield-effects */
import { put } from 'redux-saga/effects';
import { staticErrorResponse } from 'utils/globalUtils';
import authSaga, {
    getToken,
    checkAuthToken,
} from '../saga';

import {
    doLogin,
    loginSuccess,
    loginFailed,
    checkAuth,
    checkAuthDone,
    checkAuthError,
} from '../actions';

const userPayload = { username: '', password: '' };
const failedPayload = staticErrorResponse({ text: '{}' });
const successPayload = { success: true, token: '123' };

let generator;
describe('authSaga()', () => {
    generator = authSaga();

    it('should yield getToken() when LOGIN is dispatched', (done) => {
        const mock = generator.next().value;
        const expected = put(doLogin({ username: '', password: '' }));
        expect(mock.PUT).toEqual(expected.FORK);
        done();
    });

    it('should trigger checkAuthToken when AUTHENTICATE is dispatched', (done) => {
        const mock = generator.next().value;
        const expected = put(checkAuth());
        expect(mock.PUT).toEqual(expected.FORK);
        done();
    });
});

describe('getToken()', () => {
    beforeEach(() => {
        generator = getToken({ payload: userPayload });
    });
    it('should trigger loginSuccess() when success', (done) => {
        expect(generator.next()).toMatchSnapshot();
        const mock = generator.next(successPayload).value;
        const expected = put(loginSuccess(successPayload));
        expect(mock).toEqual(expected);
        done();
    });

    it('should trigger loginFailed() when API returned falsy success', (done) => {
        expect(generator.next()).toMatchSnapshot();
        const mock = generator.next(failedPayload).value;
        const expected = put(loginFailed(failedPayload));
        expect(mock).toEqual(expected);
        done();
    });

    it('should trigger loginFailed() no response returned', (done) => {
        expect(generator.next()).toMatchSnapshot();
        const errorPayload = staticErrorResponse({ text: JSON.stringify('Test Error') });
        const mock = generator.next('Test Error').value;
        const expected = put(loginFailed(errorPayload));
        expect(mock).toEqual(expected);
        done();
    });

    it('should throw loginFailed() if the generator throws error', (done) => {
        expect(generator.next()).toMatchSnapshot();
        const errorPayload = staticErrorResponse({ text: 'Error: {}' });
        const mock = generator.throw({}).value;
        const expected = put(loginFailed(errorPayload));
        expect(mock).toEqual(expected);
        done();
    });
});

describe('checkAuth()', () => {
    beforeEach(() => {
        generator = checkAuthToken();
    });

    it('should trigger checkAuthDone() when success', (done) => {
        expect(generator.next()).toMatchSnapshot();

        const mock = generator.next(successPayload.token).value;
        const expected = put(checkAuthDone(successPayload.token));
        expect(mock).toEqual(expected);
        done();
    });

    it('should throw checkAuthError() when there is no session', (done) => {
        expect(generator.next()).toMatchSnapshot();
        const mock = generator.next().value;

        const errorPayload = staticErrorResponse({ text: 'Session expired. Please login' });
        const expected = put(checkAuthError(errorPayload));
        expect(mock).toEqual(expected);
        done();
    });
});
