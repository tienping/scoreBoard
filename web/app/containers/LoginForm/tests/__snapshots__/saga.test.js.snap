// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkAuth() should throw checkAuthError() when there is no session 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`checkAuth() should trigger checkAuthDone() when success 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`getToken() should throw loginFailed() if the generator throws error 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [
        "undefined/auth/token",
        Object {
          "headers": Object {
            "authorization": "Basic Og==",
          },
          "method": "POST",
        },
      ],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`getToken() should trigger loginFailed() no response returned 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [
        "undefined/auth/token",
        Object {
          "headers": Object {
            "authorization": "Basic Og==",
          },
          "method": "POST",
        },
      ],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`getToken() should trigger loginFailed() when API returned falsy success 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [
        "undefined/auth/token",
        Object {
          "headers": Object {
            "authorization": "Basic Og==",
          },
          "method": "POST",
        },
      ],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`getToken() should trigger loginSuccess() when success 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [
        "undefined/auth/token",
        Object {
          "headers": Object {
            "authorization": "Basic Og==",
          },
          "method": "POST",
        },
      ],
      "context": null,
      "fn": [Function],
    },
  },
}
`;
